torch/_inductor/scheduler.py

```diff
@@ -2066,6 +2067,27 @@ def can_fusion_increase_peak_memory(
         )
         return proximity_score > 64
 
+    def decide_fusion_fail_reason(self, node1, node2, common_buf_names):
+        """
+        Try to decide reasons why fusion fail due to no shared memory even though
+        there are common buffers.
+        """
+        reasons = {}
+        node1_name2dep = {dep.name: dep for dep in node1.read_writes.reads_and_writes()}
+        node2_name2dep = {dep.name: dep for dep in node2.read_writes.reads_and_writes()}
+
+        for buf_name in common_buf_names:
+            lhs_dep = node1_name2dep[buf_name]
+            rhs_dep = node2_name2dep[buf_name]
+
+            if lhs_dep.get_numel() != rhs_dep.get_numel():
+                reasons[buf_name] = "different numel" 
+                continue
+
+            # Add more rules here
```

Comment:
Maybe print out the deps themselves for common bufs ?

Reply:
SchedulerNode.debug_str() already includes that.

BTW, this is not ready for review yet since i need add more rules to make it more useful. lol. But thanks for the stamp :)

Reply:
(Sent out for review by mistake..)

Reply:
> Maybe print out the deps themselves for common bufs ?

I found somethings it's more convenient to show the dep in the CSV directly so people don't need to open the text file for SchedulerNode.debug_str() to further look up. I'll do this when appropriate.